internal/operators/operators.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operators
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/inspektor-gadget/inspektor-gadget/internal/logger"
+	"github.com/inspektor-gadget/inspektor-gadget/internal/parser"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+type Runner interface {
+	ID() string
+	Parser() parser.Parser
+	Gadget() gadgets.Gadget
+	Context() context.Context
+	Operators() Operators
+	Logger() logger.Logger
+}
+
+type (
+	EnricherFunc func(any) error
+)
+
+var ErrAsync = errors.New("async")
+
+type Operator interface {
+	// Name must return a unique name for the operator
+	Name() string
+
+	// Description is an optional description to show to the user
+	Description() string
+
+	// Params will return global params (required) for this operator
+	Params() params.ParamDescs
+
+	// PerGadgetParams will return params (required) per gadget instance of the operator
+	PerGadgetParams() params.ParamDescs
```

Comment:
I had that name first as well, but it collides with the actual GadgetParams and might be confusing. In the end, it's still OperatorParams (and not GadgetParams), but can be different per gadget run.

Reply:
I'm also thinking about this name. Would be `GadgetParams()` actually confusing? (anyway, we can delay the discussion on these nits).

Reply:
I think that it still would confuse, yes. We could use OperatorInstanceParams (although we're also not really happy about the OperatorInstance name) :)

Reply:
I think it could be `GadgetParams`, and whenever we use them outside the operator context, we name the variable `operatorGadgetParams`. WDYT?

Reply:
> I think it could be `GadgetParams`, and whenever we use them outside the operator context, we name the variable operatorGadgetParams. WDYT?

I don't think switching names based on context is a good idea.

> What about `GadgetCommonParams()`?

IMHO that would still suggest that those params belong to a gadget and not to an operator.

Maybe we could integrate whatever `Runner()` is going to be called into the name, as it's tied to that entity somehow.